mnefun/_mnefun.py

```diff
@@ -1368,10 +1368,10 @@ def do_preprocessing_combined(p, subjects):
             meg, eeg = 'meg' in raw, 'eeg' in raw
             picks = pick_types(raw.info, meg=meg, eeg=eeg, eog=False,
                                exclude=[])
-            assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict
```

Comment:
That would be the more likely fix to the problem.  Initially `assert type(p.auto_bad_reject) and type(p.auto_bad_flat) == dict` failed if I didn't define `auto_bad_flat`. I think your recommendation with two assertions thoroughly covers all possible use cases.